lib/luathread.c

```diff
@@ -175,6 +177,53 @@ static int luathread_task(lua_State *L)
 	return 1;
 }
 
+/***
+* Allows the current kernel thread to handle a specific signal.
+* This function enables the thread to receive and process the specified signal by unblocking it at the kernel level.
+* @function allow
+* @tparam integer signum The signal number to allow (e.g., 15 for SIGTERM)
+* @treturn nil
+* @usage
+* local t = thread.current()
+* t:allow_signal(15)  -- Allow SIGTERM
+*/
+
+static int luathread_allow_signal(lua_State *L)
+{
+ 	int signum = luaL_checkinteger(L, 2);
+ 	allow_signal(signum);
+	
+ 	return 0;
+}
+
+static int luathread_send_signal(lua_State *L)
+{
+ 	int signum = luaL_checkinteger(L, 2);
+   	lunatik_object_t *object = lunatik_toobject(L, 1);
```

Comment:
I think you miss my point on the checker function.. take a look on the code base, you will find that we have a macro ready for this pattern

Reply:
No.. I'm referring to this https://github.com/luainkernel/lunatik/blob/270622c10aa238a5405f6f1122cd66597dcbe288/lib/luasocket.c#L122